### PR TITLE
updating docs with examples

### PR DESCRIPTION
**Pull request by choldgraf**
I wanted to wrap my head around `okgrade`, and figured the best way to do so would be to write some docs :-)

This PR:

* Adds some user-friendly sample Jupyter Notebooks in an `examples/` folder
* Adds more language in usage for the two main use-cases of `okgrade` (interactive feedback and auto-grading)
* Cleans up the language here and there for organization / clarity
* Adds Binder links so people can interact

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,6 +1,39 @@
 # okgrade
 
 `okgrade` is a small library for autograding Jupyter notebooks & python files.
+It is not dependent on any other grading packages and tools (such as `okpy` or
```
Comment (yuvipanda) [inline docs/index.md:4]: I'm unsure what using these 'alongside' means!

File: docs/index.md [hunk 2]
```diff
@@ -1,6 +1,39 @@
...
+`nbgrader`), though it may be used *alongside* them in order to facilitate the
+grading process. It performs all computation/grading on the computer that
+runs the `okgrade` code, and does not distribute notebooks or run them on a
+remote server.
+
+## Overview
+
+`okgrade` allows you to combine two kinds of files:
+
+1. Jupyter Notebooks in which students have done their work.
```
Comment (yuvipanda) [inline docs/index.md:14]: We also support plain .py files.

**Review by yuvipanda — CHANGES_REQUESTED** [review]
w00t, thank you for taking the effort to do this :)

I'm continuing to break everything, so will ping here again once all the breaking has stabilized a bit :)

File: docs/index.md (hunk 2 above)
Comment (choldgraf) [reply docs/index.md:14]: good to know, didn't realize that :-D

**Comment by choldgraf** [comment]
cool - lemme know when it's time for me to take another crack at this!

**Comment by choldgraf** [comment]
Gonna close this since GMM has changed quite a bit and I'm not sure yuvi wants to do the code reviews for this :-)

**Pull request closed without merging**